ui/src/components/FacetCard.js

```diff
@@ -1,37 +1,68 @@
-        <ListItem
-            className="listItem"
-            key={facetValue.name}
-            leftCheckbox={<Checkbox
-                onCheck={(event, isInputChecked) => props.updateFacets(facet.name, facetValue.name, isInputChecked)}
-            />}
-            primaryText={<div>
-                <div className="facetValueName">{facetValue.name}</div>
-                <div className="facetValueCount">{facetValue.count}</div>
-            </div>}/>
-    );
-    return (
-        <Card className="facetCard">
-            <div className="cardHeader">
-                <div>{facet.name}</div>
-                <div className="subHeader">
-                    <span>{totalCount}</span>
-                    <span className="numberSelected">0 / {facetValues.length}</span>
+class FacetCard extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.facetValues = this.props.facet.values;
+        this.totalCount = this.props.totalCount;
+
+        this.state = {
+            selectedValues: []
+        };
+
+        this.onValueCheck = this.onValueCheck.bind(this);
+        this.isUnselected = this.isUnselected.bind(this);
+    }
+
+    render() {
+        const facetValues = this.props.facet.values.map((facetValue) =>
+            <ListItem
+                className="listItem"
+                key={facetValue.name}
+                leftCheckbox={<Checkbox
+                    onCheck={(event, isInputChecked) => this.onValueCheck(facetValue, isInputChecked)}
+                />}
+                primaryText={<div className={this.isUnselected(facetValue) ? " grayText" : ""}>
+                    <div className="facetValueName">{facetValue.name}</div>
+                    <div className="facetValueCount">{facetValue.count}</div>
+                </div>}/>
+        );
+        return (
+            <Card className="facetCard">
+                <div className="cardHeader">
+                    <div>{this.props.facet.name}</div>
+                    <div className="subHeader">
+                        <span>{this.props.totalCount}</span>
+                        <span className="numberSelected">{this.state.selectedValues.length} / {facetValues.length}</span>
+                    </div>
                 </div>
-            </div>
-            <List>{facetValues}</List>
-        </Card>
-    );
+                <List>{facetValues}</List>
+            </Card>
+        );
+    }
+
+    /** Update this card's state and trigger the callback on the parent component. */
+    onValueCheck(facetValue, isInputChecked) {
+        let newValues = this.state.selectedValues;
+        if (isInputChecked) {
+            newValues.push(facetValue.name);
+        } else {
+            newValues.splice(newValues.indexOf(facetValue.name), 1);
+        }
+        this.setState({selectedValues: newValues});
+        this.props.updateFacets(this.props.facet.name, facetValue.name, isInputChecked);
+    }
+
+    isUnselected(facetValue) {
```

Comment:
isSelected() seems more natural to me

Reply:
The set of not unselected things actually consists of  both selected values and all values in facets which have no selected values. I chose this naming using the negation of those things in order to avoid that messiness. And functionally, we don't have any need to negate `isUnselected`, so the syntax would be `~isSelected` anyway.